src/main/java/com/server/ResponseBodyBuilder.java

```diff
@@ -53,30 +53,45 @@ public String eatCookieBody(RequestParams requestParams) {
         return "mmmm " + requestParams.getCookies().get("type");
     }
 
-    private String directoryListingBody(String path){
-        String dirPath = this.publicDir + path;
-        return getFiles(dirPath);
+    private String directoryLinksBody(RequestParams requestParams) throws IOException {
+        String dirPath = this.publicDir + requestParams.getPath();
+        String linkedFilesBody = getLinkedFiles(dirPath);
+        if(!linkedFilesBody.equals("")) {
+            return htmlBuilder().replace("$body", linkedFilesBody);
+        } else {
+            return "";
+        }
     }
 
-    private String getFiles(String dirPath){
+    private String getFileContents(File file) throws IOException {
+        FileReader fileReader = new FileReader(file);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
+        RequestReader reader = new RequestReader(bufferedReader);
+        return reader.getRequestedFileContents();
+    }
+
+    private String getLinkedFiles(String dirPath){
         StringBuilder fileNames = new StringBuilder();
         final File folder = new File(dirPath);
 
         if(folder.listFiles() != null){
             for (final File fileEntry : folder.listFiles()) {
-                fileNames.append(fileEntry.getName());
-                fileNames.append(" ");
+                if(!(fileEntry.getName().equals(".DS_Store"))){
+                    fileNames.append("<a href=\"")
```

Comment:
What do you think about extracting this link building out into its own method, kind of like `htmlBuilder()`? Maybe `htmlLinkBuilder()`?

Reply:
hmmm could be a good idea, thanks will look into it!